python/dgl/graphbolt/impl/sampled_subgraph_impl.py

```diff
@@ -68,6 +68,63 @@ def __post_init__(self):
             ), "Nodes in pairs should be of type torch.Tensor."
 
 
+def _to_reverse_ids(node_pair, reverse_row_node_ids, reverse_column_node_ids):
+    u, v = node_pair
+    if reverse_row_node_ids is not None:
+        u = reverse_row_node_ids[u]
+    if reverse_column_node_ids is not None:
+        v = reverse_column_node_ids[v]
+    return (u, v)
+
+
+def _relabel_homo_edges(lhs_edges, rhs_edges):
+    """Relabel the source nodes and destination nodes independently for two set
+    of edges."""
+    src = torch.cat([lhs_edges[0], rhs_edges[0]])
+    dst = torch.cat([lhs_edges[1], rhs_edges[1]])
+    _, src = torch.unique(src, return_inverse=True)
+    _, dst = torch.unique(dst, return_inverse=True)
+    lhs_edges = (src[: lhs_edges[0].numel()], dst[: lhs_edges[0].numel()])
+    rhs_edges = (src[lhs_edges[0].numel() :], dst[lhs_edges[0].numel() :])
+    return lhs_edges, rhs_edges
+
+
+def _exclude_homo_edges(lhs_edges, rhs_edges):
+    lhs_edges, rhs_edges = _relabel_homo_edges(lhs_edges, rhs_edges)
```

Comment:
Why do we need to relabel before we exclude? It is a little bit confusing.

Reply:
I'm also a little confused as both ids are already converted to non-compacted format. And `unique `should be a relatively costly op.

Reply:
This is to convert each edge to a single value to apply `torch.isin`. Another solution is to compact an edge `(u, v)` to `(u << 32 + v)`. But it does not work if some node id exceed `2^32`. Therefore I further compact the nodes ids of all edges to make their range smaller.